docs/proposals/kubemark.md

```diff
@@ -0,0 +1,163 @@
+
+### Elasticsearch
+
+Similarly to Heapster Elasticsearch runs outside the Master machine but generates some traffic on it. Fluentd “daemon” running on Master periodically sends Docker logs it gathered to the Elasticsearch running on one of the Nodes. In the initial version we omit Elasticsearch, as it produces only a constant small load on Master Node that does not change with the size of the cluster.
+
+## Necessary work
+
+There are three more or less independent things that needs to be worked on:
+- HollowNode implementation, creating a library/binary that will be able to listen to Watches and respond in a correct fashion with Status updates. This also involves creation
+of a CloudProvider that can produce such Hollow Nodes.
+- Kubemark setup, including figuring networking model, number of Hollow Nodes that will be allowed to run on a single “machine”, writing setup/run/teardown scripts
+- Creating a Player component that will send requests to the API server putting a load on a cluster. This involves creating a way to specify desired workload. This task is
+very well isolated from the rest, as it is about sending requests to the real API server. Because of that we can discuss requirements separately.
+
+## Concerns
+
+Network performance most likely won't be a problem for the initial version if running on bare metal or 'pure' cloud provider, as Kubemark
+will be running on standard networking stack (no cloud-provider software routes, or overlay network is needed, as we don't need custom
+routing between Pods). Similarly we don't think that running Kubemark on Kubernetes virtualized cluster networking will cause noticable
+performance impact, but it requires testing.
+
+On the other hand when adding additional features it may turn out that we need to simulate Kubernetes Pod network. In such, when running
+'pure' Kubemark we may try one of the following:
+  - running overlay network like Flannel or OVS instead of using cloud providers routes,
+  - write simple network multiplexer to multiplex communications from the Hollow Kubelets/KubeProxies on the machine.
+
+In case of Kubemark on Kubernetes it may turn that we run into a problem with adding yet another layer of network virtualization, but we
+don't need to solve this problem now.
+
+## Work plan
+
+- Teach/make sure that Master can talk to multiple Kubelets on the same Machine:
+  - make sure that Master can talk to a Kubelet on non-default port,
+  - make sure that Master can talk to all Kubelets on different ports,
+- Write HollowNode library:
+  - new HollowProxy,
+  - new HollowKubelet,
+  - new HollowNode combining the two,
+  - make sure that Master can talk to two HollowKubelets running on the same machine
+- Write a player that will automatically put some predefined load on Master, <- this is the moment when it’s possible to play with it and is useful by itself for
```

Comment:
For option #2, need to get the Hollow nodes and master components of the Hollow cluster to run on Kuberenetes.